db/version_set.cc

```diff
@@ -4118,7 +4118,7 @@ Status VersionSet::ProcessManifestWrites(
         TEST_KILL_RANDOM("VersionSet::LogAndApply:BeforeAddRecord",
                          rocksdb_kill_odds * REDUCE_ODDS2);
 #ifndef NDEBUG
-        if (batch_edits.size() > 1 && batch_edits.size() - 1 == idx) {
+        if (batch_edits.size() > 0 && batch_edits.size() - 1 == idx) {
```

Comment:
Any reason for this change?

Reply:
reworded

Reply:
a test relies on the fact that, if the edit is the last in the batch, the next sync point will be executed.

Reply:
Which test relies on triggering the following test sync point when `batch_edits.size() == 1`?

Reply:
I tried to rollback this change, and the tests pass. The only behavior change of the recent commits is that when writing to MANIFEST in `DeleteUnreferencedSstFiles`, if might not switch to a new MANIFEST. The test might fail when writing to a new MANIFEST with a single edit. Anyway, I rollbacked this change.